worker/src/main/java/gov/cms/ab2d/worker/processor/TextStreamHelperImpl.java

```diff
@@ -0,0 +1,93 @@
+package gov.cms.ab2d.worker.processor;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.*;
+import java.nio.file.Path;
+
+/**
+ * Implement a plain text stream helper
+ */
+@Slf4j
+public class TextStreamHelperImpl extends StreamHelperImpl {
+
+    /**
+     * Implement the text stream helper
+     *
+     * @param path - where to create the file
+     * @param contractNumber - the contract number
+     * @param totalBytesAllowed - the total number of bytes allowed to be written to the stream
+     * @throws FileNotFoundException - if the file can't be created
+     */
+    TextStreamHelperImpl(Path path, String contractNumber, long totalBytesAllowed, int tryLockTimeout)
+            throws FileNotFoundException {
+        super(path, contractNumber, totalBytesAllowed, tryLockTimeout);
+
+        setCurrentStream(createStream());
+    }
+
+    /**
+     * Create the next stream from the next file name
+     *
+     * @return the stream
+     * @throws FileNotFoundException if you can't create the stream
+     */
+    private OutputStream createStream() throws FileNotFoundException {
+        String fileName = getPath().toString() + "/" + createFileName();
+        File f = new File(fileName);
+        f.getParentFile().mkdirs();
+        OutputStream stream = new BufferedOutputStream(new FileOutputStream(fileName));
+        Path p = Path.of(fileName);
+        getFilesCreated().add(p);
+        return stream;
+    }
+
+    /**
+     * Write data to the correct file and iterate to the next file if it exceeds the limit
+     *
+     * @param data - the data to write
+     */
+    @Override
+    public void addData(byte[] data) {
+        tryLock(getDataFileLock());
+        if (data == null || data.length == 0) {
+            return;
+        }
```

Comment:
If this condition is ever true, the lock will never release. This `if` needs to be moved inside the `try/finally`.

Reply:
...actually, it needs to be placed right before `tryLock` instead.

Reply:
Good catch

Reply:
I just moved it above. No reason to get the lock if there is no data.